src/file.cpp

```diff
@@ -2050,6 +2050,7 @@ typedef struct _FILE_ALLOCATED_RANGE_BUFFER {
 			}
 		}
 
+#if _WIN32_WINNT >= 0x0600 // only if Windows Vista or newer
```

Comment:
Yes, that's the idea and it runs in xp, just does not compile due to the use of `FILE_INFO_BY_HANDLE_CLASS`.